test/parallel/test-timers-uncaught-exception.js

```diff
@@ -1,40 +1,22 @@
 'use strict';
-require('../common');
-var assert = require('assert');
-
-var exceptions = 0;
-var timer1 = 0;
-var timer2 = 0;
+const common = require('../common');
+const assert = require('assert');
+const errorMsg = 'BAM!';
 
 // the first timer throws...
-console.error('set first timer');
-setTimeout(function() {
-  console.error('first timer');
-  timer1++;
-  throw new Error('BAM!');
-}, 100);
+setTimeout(common.mustCall(function() {
+  throw new Error(errorMsg);
+}), 100);
 
 // ...but the second one should still run
-console.error('set second timer');
-setTimeout(function() {
-  console.error('second timer');
-  assert.equal(timer1, 1);
-  timer2++;
-}, 100);
+setTimeout(common.mustCall(function() {}), 100);
 
 function uncaughtException(err) {
-  console.error('uncaught handler');
-  assert.equal(err.message, 'BAM!');
-  exceptions++;
+  assert.strictEqual(err.message, errorMsg);
 }
-process.on('uncaughtException', uncaughtException);
 
-var exited = false;
+process.on('uncaughtException', common.mustCall(uncaughtException));
+
 process.on('exit', function() {
-  assert(!exited);
-  exited = true;
   process.removeListener('uncaughtException', uncaughtException);
```

Comment:
This will fail because what was actually registered was a function returned from `common.mustCall()`.

I don't think you can have async errors during exit though, so this should be unnecessary I think.